src/coreclr/jit/gentree.cpp

```diff
@@ -7226,6 +7226,13 @@ bool GenTree::OperMayThrow(Compiler* comp)
             assert(varTypeIsInt(AsHWIntrinsic()->GetSimdBaseType()));
             return true;
         }
+#elif defined(TARGET_ARM64)
+        // If this node is embedding another node, then check that instead.
+        if (IsEmbeddingMaskOp())
+        {
+            assert(this->AsHWIntrinsic()->GetHWIntrinsicId() == NI_Sve_ConditionalSelect);
+            return this->AsHWIntrinsic()->Op(2)->OperMayThrow(comp);
```

Comment:
What if this is user written `ConditionalSelect` and contains something that throws in `op1` or `op3`? Should we check for those as well?

Reply:
> I would rather see us introducing a new variant of `OperMayThrow`

Done. Not keen on the new name (`OperOrEmbeddedChildrenMayThrow()`)

> that takes containment into account and which may be used in the rare case that the backend needs it. It should not need hardcoding to a specific oper type like this.

Specifically, it still checks IsEmbeddingMaskOp() and not containment (as there is a period where the nodes are not yet contained).

> What if this is user written `ConditionalSelect` and contains something that throws in `op1` or `op3`? Should we check for those as well?

Also done.

Reply:
> Specifically, it still checks IsEmbeddingMaskOp() and not containment (as there is a period where the nodes are not yet contained).

The liveness that is using this runs after lowering. After lowering, all the containment checks should be done. If we are not properly marking these operands as contained during lowering then that seems like a bug. Have we actually introduced another concept of "containment" called "embedding" here?

Reply:
The way I interpreted [this comment](https://github.com/dotnet/runtime/pull/115566#discussion_r2103087325) was that I shouldn't be relying on the changes that happen in lowering. Re-reading now I'm not sure if I still agree.

I'm happy to remove the new `GTF_HW_EM_OP` flag and instead `OperOrEmbeddedChildrenMayThrow()` would check nodes based on containing.

??

Reply:
> I'm happy to remove the new `GTF_HW_EMBEDDING_OP` flag and instead `OperOrEmbeddedChildrenMayThrow()` would check nodes based on containing.

Fixed it up to work that way

Reply:
>  I think @tannergooding's comment was about the same thing

Yep. There's quite a bit of nuance between LIR and HIR in some cases and the guarantees we have in place. We generally don't want to make modifications to general IR for something that is LIR only and instead want to have a `Lowering::*` specific function instead -or- try to work with what LIR may expect, such as by introducing a different LIR specific intrinsic ID (which is what we do for `xarch` in several cases).